File: 347-top-k-frequent-elements/top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        map<int,int>m;
        for(auto num: nums)
        {
            m[num]++;
        }
        vector<int>ans;
        priority_queue<pair<int,int>>pq;
        for(auto i : m)
        {
            pq.push({ i.second , i.first,});
        }
        while(k--)
        {
            int count = pq.top().first;
            int num = pq.top().second;
            cout<<num<<"------->"<<count<<endl;
            pq.pop();
            ans.push_back(num);
        }
        return ans;
    }
};

File: 347-top-k-frequent-elements/top-k-frequent-elements.py
import heapq
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        s = {}
        for n in nums:
            s[n] = s.get(n, 0) + 1
        h = []
        for n in s:
            heapq.heappush(h, (-1 *s[n], n))
        res = []
        if k > len(s):
            return []
        for i in range(k):
            t = heapq.heappop(h)
            res.append(t[1])
        return res